neqo-crypto/src/agent.rs

```diff
@@ -571,6 +675,30 @@ impl SecretAgent {
         })
     }
 
+    /// Install a certificate compression mechanism.
+    ///
+    /// # Errors
+    /// If the compression mechanism with the same id is already registered
+    /// If too many compression mechanisms are already registered
+    ///
+    /// This returns an error if the certificate compression could not be established
+    ///
+    /// [RFC8879]: https://datatracker.ietf.org/doc/rfc8879/
+    pub fn set_certificate_compression<T: SafeCertificateCompression>(&mut self) -> Res<()> {
+        if T::ID == 0 {
+            return Err(Error::InvalidCertificateCompressionID);
+        }
+        let compressor: ssl::SSLCertificateCompressionAlgorithm =
+            ssl::SSLCertificateCompressionAlgorithm {
+                id: T::ID,
+                #[expect(clippy::as_ptr_cast_mut, reason = "requires to be const char")]
+                name: T::NAME.as_ptr() as *mut ::std::os::raw::c_char,
+                encode: T::ENABLE_ENCODING.then_some(<T as UnsafeCertCompression>::encode_callback),
+                decode: Some(<T as UnsafeCertCompression>::decode_callback),
```

Comment:
Same here, though the raw pointer works.  (Again, untested.)

```suggestion
                decode: <T as UnsafeCertCompression>::decode_callback,
```

Reply:
I'm not 100% sure? 
```
pub decode: ::std::option::Option<
        unsafe extern "C" fn(... ) -> SECStatus,
    >,
```

The compressor requires both encoder and decoder to be Option. Or it's not what you meant?

Reply:
Oh, my bad.  I had no idea that this was indeed an `Option` in the bindings.  Forget this.  (I assumed that the pointer could just be null and that would be OK, but here we have an option.  The wonders never cease.)